client/src/main/java/org/evosuite/ga/metaheuristics/TestSuiteAdapter.java

```diff
@@ -312,28 +325,28 @@ final protected void calculateFitness(TestSuiteChromosome c)
     }
 
     @Override
-    final protected void calculateFitnessAndSortPopulation() throws UnsupportedOperationException {
-        throw new UnsupportedOperationException("not implemented");
+    final protected void calculateFitnessAndSortPopulation(){
+        algorithm.calculateFitnessAndSortPopulation();
     }
 
     @Override
-    final public int getPopulationSize() throws UnsupportedOperationException {
-        throw new UnsupportedOperationException("not implemented");
+    final public int getPopulationSize() {
+        return algorithm.getPopulationSize();
     }
 
     @Override
-    final protected List<TestSuiteChromosome> elitism() throws UnsupportedOperationException {
-        throw new UnsupportedOperationException("not implemented");
+    final protected List<TestSuiteChromosome> elitism() {
+        return algorithm.elitism().stream().map(TestChromosome::toSuite).collect(toList());
     }
 
     @Override
-    final protected List<TestSuiteChromosome> randomism() throws UnsupportedOperationException {
-        throw new UnsupportedOperationException("not implemented");
+    final protected List<TestSuiteChromosome> randomism() {
+        return algorithm.randomism().stream().map(TestChromosome::toSuite).collect(Collectors.toList());
```

Comment:
Here, we have to ask ourselves the same question as for `elitism()` (see above). In any case, whatever the outcome will be, we have to make sure that we handle `elitism()` and `randomism()` consistently.

Reply:
Since it has just been decided to keep your implementation of `elitism()` we keep it here as well